gotime.js

```diff
@@ -1078,11 +1099,35 @@ var fmtParts = {
     return formatTimestampTokens(timestamp, tokenizeFormat(format));
   }
 
+  // Formats an object containing date and time components using the provided format
+  // @param {Object} dateTime Date and time of the form: { date: {year: 2017, month: 10, day: 27}, time: {hour: 14, minute: 15, second: 30}  }
+  // @param {string} format
+  function formatDateTime(dateTime, format) {
```

Comment:
@theothertomelliott It might be that this is a little too Pages specific.  In practice, we wrap this library with a stub in our Soy2JS binding layer.  Perhaps that layer can be responsible for unpacking the object and calling `formatDateTimeTokens()`

Reply:
I'd originally implemented this as a set of positional params like formatDateTimeTokens, but Chris W suggested providing an object with named params as a more idiomatic JS implementation. Then it made sense to use the form we use for profiles.

Reply:
Hmm, I guess, its just a strange structure (that isn't found elsewhere in JS).

Reply:
I don't feel super strongly since we're likely the only consumers of this library.